packages/core/utils/src/event-bus/index.ts

```diff
@@ -24,6 +24,15 @@ export abstract class AbstractEventBusModuleService
   abstract emit<T>(data: EventBusTypes.EmitData<T>[]): Promise<void>
   abstract emit<T>(data: EventBusTypes.Message<T>[]): Promise<void>
 
+  abstract stageEvent<TData = unknown>(
+    eventGroupId: string,
+    eventName: string,
+    data: TData
+  ): Promise<void>
+
+  abstract releaseStagedEvents(eventGroupId: string): Promise<void>
+  abstract clearStagedEvents(eventGroupId: string): Promise<void>
```

Comment:
Maybe we can find better names here?
I like only `release` and `dismiss`
or `releaseGroupedEvents` and `cancelGroupedEvents`/`dismissGroupedEvents`

Reply:
can used grouped instead, but wrt to a queue, the action words are release and clear right?

Reply:
resolved over chat - decision to use group, but the action words the same 💪🏻